paper.md

```diff
@@ -419,12 +418,29 @@ Popper falls in this category of container-native workflow engines but it does n
 Cloud-native computing is an emerging paradigm in software development that aims to utilize existing cloud infrastructure to build, run, and scale applications in the cloud.
 The cloud-native paradigm is a subset of the container-native paradigm since it encourages running applications not only inside containers but also on cloud infrastructure.
 Container engines like Docker and orchestration tools like Kubernetes have become an integral part of the cloud-native paradigm over the years.
-With the wide-spread acceptance of the cloud-native paradigm, several cloud-native workflow engines like  Argo, Pachyderm, Brigade have come into existence.
+With the wide-spread acceptance of the cloud-native paradigm, several cloud-native workflow engines like  Argo [@argocommunity_argoproj_2019], Pachyderm [@novella_containerbased_2018], Brigade [@brigade] have come into existence.
 These workflow engines facilitate running workflows in the cloud by running an entire workflow in containers managed by Kubernetes.
 The limitation of these workflow engines is the requirement of having access to a Kubernetes cluster which can block users from running workflows in the absence of one.
 Although Popper can run workflows in the cloud using Kubernetes, it does not necessarily require access to a Kubernetes cluster for running the containerized steps of a workflow.
 Popper is not exclusively cloud-native since it does not assume the presence of a Kubernetes cluster for running workflows, but in addition to being able to work as cloud-native i.e. run workflows on Kubernetes, it can also behave as container-native in different computing environments like a local machine, Slurm and cloud VM instances over SSH.
 
+## Continuous Integration Tools
+
+Continuous Integration is a DevOps practice that enables building and testing code frequently to prevent the accumulation of broken code and support faster development cycles.
+Tools like Travis, Circle, Jenkins, GitLab-CI, etc. have become the standard for doing CI, but they were primarily built for running on the Cloud.
+This results in increased iteration time, especially for quick build and test scenarios, where the time spent in booting of VMs, scheduling of CI jobs becomes an overhead.
+Reproducing experiments on the local machine that originally run on CI, requires manually setting up the entire development environment, making the entry barrier high.
+Running on CI tools hosted locally requires knowledge and expertise in deploying and using the specific tools.
```

Comment:
we can mention a specific example, for example the [gitlab-ci runner](https://gitlab.com/gitlab-org/gitlab-runner)